panel/tests/widgets/test_slider.py

```diff
@@ -156,6 +156,68 @@ def test_date_slider(document, comm):
         assert widget.value == 1620777600000
 
 
+@pytest.mark.parametrize("start", [date(2018, 9, 1), datetime(2018, 9, 1)])
+@pytest.mark.parametrize("end", [date(2018, 9, 10), datetime(2018, 9, 10)])
+@pytest.mark.parametrize("value", [date(2018, 9, 4), datetime(2018, 9, 4)])
+def test_datetime_slider(document, comm, value, start, end):
+    datetime_slider = DatetimeSlider(
+        name='DatetimeSlider',
+        value=value,
+        start=start,
+        end=end,
+    )
+    assert datetime_slider.start == start
+    assert datetime_slider.end == end
+    assert datetime_slider.value == value
+
+    widget = datetime_slider.get_root(document, comm=comm)
+
+    assert isinstance(widget, datetime_slider._widget_type)
+    assert widget.title == 'DatetimeSlider'
+    assert widget.value == 1536019200000
+    assert widget.start == 1535760000000.0
+    assert widget.end == 1536537600000.0
+
+    epoch = datetime(1970, 1, 1)
+    widget.value = (datetime(2018, 9, 3)-epoch).total_seconds()*1000
```

Comment:
```suggestion
    epoch = datetime(1970, 1, 1)
    epoch_time = lambda x: (x-epoch).total_seconds()*1000
    widget.value = epoch_time(datetime(2018, 9, 3))
```

Also would need changes other places